apis/v1alpha1/tlsconfig_types.go

```diff
@@ -65,6 +65,14 @@ type TLSConfig struct {
 	//
 	// +required
 	CertificateRef CertificateObjectReference `json:"certificateRef,omitempty"`
+
+	// AllowedInRoute dictates if TLS certificates can be configured via Routes
+	// or not. If set to true, a TLS certificate for a hostname defined in
+	// Route takes takes precedence over the certificate defined in Gateway.
+	// +kubebuilder:default=false
+	// +optional
+	AllowedInRoute bool `json:"allowedInRoute,omitempty"`
```

Comment:
Is there a more specific name that evokes a better idea of what this field does? Perhaps `AllowRouteCertificates` or `ImportRouteCertificates`? Or maybe this needs to express that TLS config in general can be imported (merged?) from the routes?

Here's a YAML example that shows the problem:
```yaml
tls:
  mode: Terminate
  allowedInRoute: true
  certificateRef:
    name: foo
```

Reply:
nit: s/Route takes takes precedence/Route takes precedence/

Reply:
As discussed in the office hours, renamed the field to `AllowOverrideViaRoute`.